platform/lib/build/samplesBuilder.js

```diff
@@ -570,6 +584,23 @@ class SamplesBuilder {
       }
     });
 
+    // Store the sample by it's used component to show all samples for a specific
```

Comment:
A method that is named get... should never modify the state.
I know the code snipped above did already modify the parsedSample which is also bad.
I think we should refactor this.
Maybe call some methods in _createDocumentation that do the changes and then call getTeaserData with the usedComponents.

Reply:
ping